lib/engine/src/tunables.rs

```diff
@@ -19,14 +21,28 @@ pub trait Tunables {
     fn table_style(&self, table: &TableType) -> TableStyle;
 
     /// Create a memory given a memory type
+    ///
+    /// `vm_definition_location` should either point to a valid memory location
+    /// in VM memory or be `None` to indicate that the metadata for the memory
+    /// should be owned by the host.
     fn create_memory(
         &self,
         ty: &MemoryType,
         style: &MemoryStyle,
+        vm_definition_location: Option<NonNull<VMMemoryDefinition>>,
```

Comment:
If `VMMemoryDefinition` is now owned by the create_memory, how it will be Null?

Reply:
I tried to address that in the comment above:

```Rust
    /// `vm_definition_location` should either point to a valid memory location
    /// in VM memory or be `None` to indicate that the metadata for the memory
    /// should be owned by the host.
```

The idea is that `None` means that it should be owned by the thing implementing the `Memory` trait itself.  Let me know if you think there's a clearer way to word that

Reply:
I tried to address that in the comment above:

```Rust
    /// `vm_definition_location` should either point to a valid memory location
    /// in VM memory or be `None` to indicate that the metadata for the memory
    /// should be owned by the host.
```

The idea is that `None` means that it should be owned by the thing implementing the `Memory` trait itself.  Let me know if you think there's a clearer way to word that